network/p2p/peerstore/utils.go

```diff
@@ -49,3 +52,18 @@ func PeerInfoFromAddr(addr string) (*peer.AddrInfo, error) {
 	}
 	return info, nil
 }
+
+// PeerInfoFromDomainPort converts a string of the form domain:port to AddrInfo
+func PeerInfoFromDomainPort(domainPort string) (*peer.AddrInfo, error) {
+	parts := strings.Split(domainPort, ":")
```

Comment:
do you need to check for empty parts as well? Or will `multiaddr.NewMultiaddr` spot such a problem?